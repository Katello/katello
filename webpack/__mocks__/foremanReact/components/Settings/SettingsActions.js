export const loadSetting = () => { 'url'; };

export default { loadSetting };

