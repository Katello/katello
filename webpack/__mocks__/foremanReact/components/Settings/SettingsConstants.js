export const GET_SETTING_SUCCESS = 'GET_SETTING_SUCCESS';
export default GET_SETTING_SUCCESS;
