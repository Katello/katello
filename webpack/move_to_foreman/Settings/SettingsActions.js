import { foremanApi as api } from '../../services/api';

import {
  GET_SETTING_REQUEST,
  GET_SETTING_SUCCESS,
  GET_SETTING_FAILURE,
} from './SettingsConstants';

export const loadSetting = settingName => (dispatch) => {
  dispatch({ type: GET_SETTING_REQUEST });

  return api
    .get(`/settings/${settingName}`)
    .then(({ data }) => {
      dispatch({
        type: GET_SETTING_SUCCESS,
        response: data,
      });
    })
    .catch((result) => {
      dispatch({
        type: GET_SETTING_FAILURE,
        result,
      });
    });
};

export default loadSetting;
