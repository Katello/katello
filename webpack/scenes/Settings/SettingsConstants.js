export const AUTOSEARCH_DELAY = 'autosearch_delay';
export const AUTOSEARCH_WHILE_TYPING = 'autosearch_while_typing';
export const CONTENT_DISCONNECTED = 'content_disconnected';
