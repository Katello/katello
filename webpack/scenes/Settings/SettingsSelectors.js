const selectTableSettings = (state, tableName) =>
  state.katello.settings?.tables[tableName] || undefined;

export default selectTableSettings;
