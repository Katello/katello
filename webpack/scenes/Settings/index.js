import { combineReducers } from 'redux';
import tables from './Tables/TableReducer';
import settings from './SettingsReducer';

export default combineReducers({ tables, settings });
