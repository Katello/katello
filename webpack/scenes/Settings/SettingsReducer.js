import Immutable from 'seamless-immutable';
import { GET_SETTING_SUCCESS } from 'foremanReact/components/Settings/SettingsConstants';
import {
  AUTOSEARCH_DELAY,
  AUTOSEARCH_WHILE_TYPING,
  CONTENT_DISCONNECTED,
} from './SettingsConstants';

export const initialSettingsState = Immutable({
  autoSearchEnabled: true,
  autoSearchDelay: 500,
});

export default (state = initialSettingsState, action) => {
  switch (action.type) {
  case GET_SETTING_SUCCESS: {
    const { name, value } = action.response;
    switch (name) {
    case AUTOSEARCH_DELAY:
      return state.set('autoSearchDelay', value);
    case AUTOSEARCH_WHILE_TYPING:
      return state.set('autoSearchEnabled', value);
    case CONTENT_DISCONNECTED:
      return state.set('disconnected', value);
    default:
      return state;
    }
  }

  default:
    return state;
  }
};
