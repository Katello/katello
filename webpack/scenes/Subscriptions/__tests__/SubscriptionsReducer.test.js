import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import { GET_SETTING_SUCCESS } from 'foremanReact/components/Settings/SettingsConstants';

import {
  SUBSCRIPTIONS_REQUEST,
  SUBSCRIPTIONS_SUCCESS,
  SUBSCRIPTIONS_FAILURE,
  SUBSCRIPTIONS_COLUMNS_REQUEST,
  UPDATE_SUBSCRIPTION_COLUMNS,
  SUBSCRIPTIONS_QUANTITIES_REQUEST,
  SUBSCRIPTIONS_QUANTITIES_SUCCESS,
  SUBSCRIPTIONS_QUANTITIES_FAILURE,
  UPDATE_QUANTITY_REQUEST,
  UPDATE_QUANTITY_SUCCESS,
  UPDATE_QUANTITY_FAILURE,
  DELETE_SUBSCRIPTIONS_REQUEST,
  DELETE_SUBSCRIPTIONS_SUCCESS,
  DELETE_SUBSCRIPTIONS_FAILURE,
  SUBSCRIPTIONS_UPDATE_SEARCH_QUERY,
  SUBSCRIPTIONS_OPEN_DELETE_MODAL,
  SUBSCRIPTIONS_CLOSE_DELETE_MODAL,
  SUBSCRIPTIONS_OPEN_TASK_MODAL,
  SUBSCRIPTIONS_CLOSE_TASK_MODAL,
  SUBSCRIPTIONS_DISABLE_DELETE_BUTTON,
  SUBSCRIPTIONS_ENABLE_DELETE_BUTTON,
  TASK_BULK_SEARCH_SUCCESS,
  GET_TASK_SUCCESS,
  RESET_TASKS,
} from '../SubscriptionConstants';
import reducer from '../SubscriptionReducer';

jest.mock('foremanReact/components/Settings/SettingsConstants');
const fixtures = {
  'should return the initial state': {},
  'should handle SUBSCRIPTIONS_REQUEST': {
    action: {
      type: SUBSCRIPTIONS_REQUEST,
    },
  },
  'should handle UPDATE_QUANTITY_REQUEST': {
    action: {
      type: UPDATE_QUANTITY_REQUEST,
    },
  },
  'should handle DELETE_SUBSCRIPTIONS_REQUEST': {
    action: {
      type: DELETE_SUBSCRIPTIONS_REQUEST,
    },
  },
  'should handle SUBSCRIPTIONS_COLUMNS_REQUEST': {
    action: {
      type: SUBSCRIPTIONS_COLUMNS_REQUEST,
      payload: {
        tableColumns: ['col1', 'col2', 'col3'],
      },
    },
  },
  'should handle UPDATE_SUBSCRIPTION_COLUMNS': {
    action: {
      type: UPDATE_SUBSCRIPTION_COLUMNS,
      payload: {
        enabledColumns: ['col1', 'col2'],
      },
    },
  },
  'should handle SUBSCRIPTIONS_SUCCESS': {
    action: {
      type: SUBSCRIPTIONS_SUCCESS,
      response: {
        page: 1,
        per_page: 10, // eslint-disable-line camelcase
        subtotal: 20,
        results: 'some-results',
      },
      search: 'some search',
    },
  },
  'should handle DELETE_SUBSCRIPTIONS_SUCCESS': {
    action: {
      type: DELETE_SUBSCRIPTIONS_SUCCESS,
    },
  },
  'should handle UPDATE_QUANTITY_SUCCESS': {
    action: {
      type: UPDATE_QUANTITY_SUCCESS,
    },
  },
  'should handle SUBSCRIPTIONS_FAILURE': {
    action: {
      type: SUBSCRIPTIONS_FAILURE,
    },
  },
  'should handle UPDATE_QUANTITY_FAILURE': {
    action: {
      type: UPDATE_QUANTITY_FAILURE,
    },
  },
  'should handle DELETE_SUBSCRIPTIONS_FAILURE': {
    action: {
      type: DELETE_SUBSCRIPTIONS_FAILURE,
    },
  },
  'should handle SUBSCRIPTIONS_QUANTITIES_REQUEST': {
    action: {
      type: SUBSCRIPTIONS_QUANTITIES_REQUEST,
    },
  },
  'should handle SUBSCRIPTIONS_QUANTITIES_SUCCESS': {
    action: {
      type: SUBSCRIPTIONS_QUANTITIES_SUCCESS,
      payload: 'some-quantities-data',
    },
  },
  'should handle SUBSCRIPTIONS_QUANTITIES_FAILURE': {
    action: {
      type: SUBSCRIPTIONS_QUANTITIES_FAILURE,
    },
  },
  'should handle SUBSCRIPTIONS_UPDATE_SEARCH_QUERY': {
    action: {
      type: SUBSCRIPTIONS_UPDATE_SEARCH_QUERY,
      payload: 'some-query',
    },
  },
  'should handle SUBSCRIPTIONS_OPEN_DELETE_MODAL': {
    action: {
      type: SUBSCRIPTIONS_OPEN_DELETE_MODAL,
    },
  },
  'should handle SUBSCRIPTIONS_CLOSE_DELETE_MODAL': {
    action: {
      type: SUBSCRIPTIONS_CLOSE_DELETE_MODAL,
    },
  },
  'should handle SUBSCRIPTIONS_OPEN_TASK_MODAL': {
    action: {
      type: SUBSCRIPTIONS_OPEN_TASK_MODAL,
    },
  },
  'should handle SUBSCRIPTIONS_CLOSE_TASK_MODAL': {
    action: {
      type: SUBSCRIPTIONS_CLOSE_TASK_MODAL,
    },
  },
  'should handle SUBSCRIPTIONS_DISABLE_DELETE_BUTTON': {
    action: {
      type: SUBSCRIPTIONS_DISABLE_DELETE_BUTTON,
    },
  },
  'should handle SUBSCRIPTIONS_ENABLE_DELETE_BUTTON': {
    action: {
      type: SUBSCRIPTIONS_ENABLE_DELETE_BUTTON,
    },
  },
  'should handle GET_SETTING_SUCCESS with content_disconnected response': {
    action: {
      type: GET_SETTING_SUCCESS,
      response: {
        name: 'content_disconnected',
        value: 'some-value',
      },
    },
  },
  'should handle GET_SETTING_SUCCESS without content_disconnected response': {
    action: {
      type: GET_SETTING_SUCCESS,
      response: {
        name: 'some-name',
        value: 'some-value',
      },
    },
  },
  'should handle TASK_BULK_SEARCH_SUCCESS': {
    action: {
      type: TASK_BULK_SEARCH_SUCCESS,
      response: {
        results: ['result1', 'result2'],
      },
    },
  },
  'should handle GET_TASK_SUCCESS': {
    action: {
      type: GET_TASK_SUCCESS,
      response: {
        results: 'some-result',
      },
    },
  },
  'should handle RESET_TASKS': {
    action: {
      type: RESET_TASKS,
      response: {
        results: 'some-result',
      },
    },
  },
};

describe('Subscriptions reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
