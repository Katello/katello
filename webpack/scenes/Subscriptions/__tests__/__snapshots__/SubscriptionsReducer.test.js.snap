// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscriptions reducer should handle DELETE_SUBSCRIPTIONS_FAILURE 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": false,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle DELETE_SUBSCRIPTIONS_REQUEST 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle DELETE_SUBSCRIPTIONS_SUCCESS 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": false,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle GET_SETTING_SUCCESS with content_disconnected response 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": "some-value",
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle GET_SETTING_SUCCESS without content_disconnected response 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle GET_TASK_SUCCESS 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle RESET_TASKS 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_CLOSE_DELETE_MODAL 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_CLOSE_MANIFEST_MODAL 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_CLOSE_TASK_MODAL 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_COLUMNS_REQUEST 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [
    "col1",
    "col2",
    "col3",
  ],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_DISABLE_DELETE_BUTTON 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_ENABLE_DELETE_BUTTON 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": false,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_FAILURE 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": false,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_OPEN_DELETE_MODAL 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": true,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_OPEN_MANIFEST_MODAL 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": true,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_OPEN_TASK_MODAL 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": true,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_QUANTITIES_FAILURE 1`] = `
Object {
  "availableQuantities": Object {},
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_QUANTITIES_REQUEST 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": true,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_QUANTITIES_SUCCESS 1`] = `
Object {
  "availableQuantities": "some-quantities-data",
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_REQUEST 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_SUCCESS 1`] = `
Object {
  "activePermissions": Object {
    "can_delete_manifest": undefined,
    "can_edit_organizations": undefined,
    "can_import_manifest": undefined,
    "can_manage_subscription_allocations": undefined,
  },
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 20,
  "loading": false,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 1,
    "perPage": 10,
  },
  "quantitiesLoading": false,
  "results": "some-results",
  "search": "some search",
  "searchIsActive": true,
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle SUBSCRIPTIONS_UPDATE_SEARCH_QUERY 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "some-query",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle TASK_BULK_SEARCH_SUCCESS 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle UPDATE_QUANTITY_FAILURE 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": false,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle UPDATE_QUANTITY_REQUEST 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle UPDATE_QUANTITY_SUCCESS 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": false,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should handle UPDATE_SUBSCRIPTION_COLUMNS 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [
    "col1",
    "col2",
  ],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;

exports[`Subscriptions reducer should return the initial state 1`] = `
Object {
  "availableQuantities": null,
  "deleteButtonDisabled": true,
  "deleteModalOpened": false,
  "disconnected": false,
  "itemCount": 0,
  "loading": true,
  "manifestModalOpened": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "quantitiesLoading": false,
  "results": Array [],
  "searchQuery": "",
  "selectedTableColumns": Array [],
  "tableColumns": Array [],
  "taskModalOpened": false,
  "tasks": Array [],
}
`;
